src/common/options/CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+set(common_options_srcs build_options.cc)
+
+# to mimic the behavior of file(CONFIGURE ...)
+file(GENERATE OUTPUT configure_file.cmake
+  CONTENT "configure_file(\${input_file} \${output_file} @ONLY)")
+function(file_configure input_file output_file)
+  set(cmake_defs
+    -D input_file=${input_file}
+    -D output_file=${output_file})
+  file(STRINGS ${input_file} subvars REGEX "@[^@]+@")
+  foreach(line ${subvars})
+    string(REGEX REPLACE ".*@([^@]+)@.*" "\\1"
+      var "${line}")
+    set(value ${${var}})
+    list(APPEND cmake_defs -D ${var}=${value})
+  endforeach()
+  add_custom_command(OUTPUT ${output_file}
+    COMMAND ${CMAKE_COMMAND} ${cmake_defs} -P configure_file.cmake
+    DEPENDS ${input_file}
+    VERBATIM)
+endfunction()
+
+function(add_options name)
+  set(yaml_in_file ${CMAKE_CURRENT_SOURCE_DIR}/${name}.yaml.in)
+  set(yaml_file ${CMAKE_CURRENT_BINARY_DIR}/${name}.yaml)
+  file_configure("${yaml_in_file}"
+    "${yaml_file}" @ONLY)
+  set(src_file "${name}_options.cc")
+  add_custom_command(
+    OUTPUT ${src_file}
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/y2c.py
+      --input ${yaml_file}
+      --output ${src_file}
+      --name ${name}
+    DEPENDS ${yaml_file})
+  list(APPEND common_options_srcs
+    ${src_file})
+  set(common_options_srcs ${common_options_srcs} PARENT_SCOPE)
+endfunction()
+
+set(osd_erasure_code_plugins "jerasure" "lrc")
+if(WITH_EC_ISA_PLUGIN)
+  list(APPEND osd_erasure_code_plugins "isa")
+endif()
+string(REPLACE ";" " " osd_erasure_code_plugins "${osd_erasure_code_plugins}")
```

Comment:
where does this cmake variable get applied to the configuration?

this looks related to a new vstart.sh issue where the mon fails to start with:
> load: jerasure load: lrc load dlopen(build/lib/libec_isa.so): build/lib/libec_isa.so: cannot open shared object file: No such file or directory

ceph-mon seems to be taking `osd_erasure_code_plugins` directly from the yaml (`default: jerasure lrc isa`) and tries to load `isa` even though `WITH_EC_ISA_PLUGIN` is disabled

Reply:
@cbodley thank you for pointing out this regression. should be fixed by #40971 .